src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -2239,6 +2246,19 @@ bool CanDeleteSecretKey()
             return await SendToLeaderAsync(command);
         }
 
+        private async Task UpdateGenAiStateToLastEtag(string databaseName, GenAiConfiguration genAiConfiguration)
+        {
+            // Update the ETL process state with the last etag and change vector so that GenAI task will start from new docs only
+
+            var database = await DatabasesLandlord.TryGetOrCreateResourceStore(databaseName);
+            (long lastEtag, string lastCv) = database.ReadLastEtagAndChangeVector();
+            
+            var updateStateCmd = new UpdateEtlProcessStateCommand(databaseName, genAiConfiguration.Name, genAiConfiguration.Transforms[0].Name, lastEtag, lastCv, NodeTag,
+                LicenseManager.HasHighlyAvailableTasks(), database.DbBase64Id, RaftIdGenerator.NewId(), skippedTimeSeriesDocs: null, lastBatchTime: null);
```

Comment:
How does this works? Do we even have the command here yet? Because it is called before `AddGenAiCommand` is created.

Also - why aren't we just setting the ` (long lastEtag, string lastCv)` in the  `AddGenAiCommand` ?